server/src/grpc/remote_engine_service/mod.rs

```diff
@@ -84,35 +86,40 @@ impl StreamReadReqKey {
     }
 }
 
-struct ExecutionGuard<F: FnMut()> {
-    f: F,
-    cancelled: bool,
+pub type StreamReadRequestNotifiers =
+    Arc<RequestNotifiers<StreamReadReqKey, mpsc::Sender<Result<RecordBatch>>>>;
+pub type BoxedStream<T> = Box<dyn Stream<Item = T> + Send + Unpin>;
+
+struct ReceiverStreamWrapper {
+    inner: BoxedStream<Result<RecordBatch>>,
```

Comment:
Shall we define it as a generic params? I guess there is no need to make it as trait object.